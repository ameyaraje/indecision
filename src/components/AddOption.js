import React from 'react';

export default class AddOption extends React.Component {
    state = {
        errorMessage: undefined
    };

    addOption = (e) => {
        e.preventDefault();

        const option = e.target.elements.option.value.trim();
        const error = this.props.handleAddOption(option);

        this.setState(() => {
            return {
                errorMessage: error
            };
        });

        if (!error) {
            e.target.elements.option.value = '';
        }
        
    }

    render() {
        return (
            <div>
                {this.state.errorMessage && <p className="empty-option">{this.state.errorMessage}</p>}
                <form className="add-new-option" onSubmit={this.addOption}>
                    <input className="add-new-option__input" type="text" name="option"></input>
                    <button className="button">Add Option</button>
                </form>
            </div>
        );
    }
}