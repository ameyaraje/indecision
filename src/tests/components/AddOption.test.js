import React from 'react';
import { shallow } from 'enzyme';

import AddOption from '../../components/AddOption';

test('should render AddOption component', () => {
    const wrapper = shallow(<AddOption />);

    expect(wrapper).toMatchSnapshot();
});