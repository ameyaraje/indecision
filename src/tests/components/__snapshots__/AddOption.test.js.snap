// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddOption component 1`] = `
<div>
  <form
    className="add-new-option"
    onSubmit={[Function]}
  >
    <input
      className="add-new-option__input"
      name="option"
      type="text"
    />
    <button
      className="button"
    >
      Add Option
    </button>
  </form>
</div>
`;
